src/Bridge/Azure/OpenAI/PlatformFactory.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpLlm\LlmChain\Bridge\Azure\OpenAI;
+
+use PhpLlm\LlmChain\Bridge\OpenAI\Embeddings;
+use PhpLlm\LlmChain\Bridge\OpenAI\GPT;
+use PhpLlm\LlmChain\Platform;
+use Symfony\Component\HttpClient\EventSourceHttpClient;
+
+final readonly class PlatformFactory
+{
+    public static function create(
+        string $baseUrl,
+        string $deployment,
+        string $apiVersion,
+        #[\SensitiveParameter] string $apiKey,
+    ): Platform {
+        $httpClient = new EventSourceHttpClient();
+        $embeddingsResponseFactory = new EmbeddingsResponseFactory($httpClient, $baseUrl, $deployment, $apiVersion, $apiKey);
+        $GPTResponseFactory = new GPTResponseFactory($httpClient, $baseUrl, $deployment, $apiVersion, $apiKey);
+
+        return new Platform(
+            [$GPTResponseFactory, $embeddingsResponseFactory],
+            [new GPT\ResponseConverter(), new Embeddings\ResponseConverter()],
+        );
```

Comment:
one important benefit:
response factory for the Azure specific platform is different, but response factory can sit on the openai default one